test/specs/search/SearchService.spec.js

```diff
@@ -1,69 +1,195 @@
+      getCoordinate(extent, '2600 987.2 1200 556.5').then(function(position) {
+        expect(position).to.eql([600987.2, 200556.5]);
+        expect(spy.callCount).to.eql(1);
+        done();
+      });
+      $httpBackend.flush();
+      $rootScope.$digest();
     });
 
-    it('supports latitude and longitude as decimal', function() {
-      expect(getCoordinate(extent, '6.96948 46.9712')).to.eql([564298.937, 202343.701]);
-      expect(getCoordinate(extent, '46.9712 6.96948')).to.eql([564298.937, 202343.701]);
+    it('supports latitude and longitude as decimal', function(done) {
+      var spy = sinon.spy(gaReframe, 'get95To03')
+      getCoordinate(extent, '6.96948 46.9712').then(function(position) {
+        expect(position).to.eql([564298.937, 202343.701]);
+        expect(spy.callCount).to.eql(0);
+        done();
+      });
+      $rootScope.$digest();
+      getCoordinate(extent, '46.9712 6.96948').then(function(position) {
+        expect(position).to.eql([564298.937, 202343.701]);
+        expect(spy.callCount).to.eql(0);
+        done();
+      });
+      $rootScope.$digest();
     });
 
-    it('supports latitude and longitude as DMS', function() {
-      expect(getCoordinate(extent, '7° E 46° N')).to.eql([566016.05, 94366.859]);
-      expect(getCoordinate(extent, '7° 1\' E 46° N')).to.eql([567307.273, 94359.756]);
-      expect(getCoordinate(extent, '7° 1\' 25.0\'\' E 46° N')).to.eql([567845.283, 94356.877]);
-      expect(getCoordinate(extent, '7° 1\' 25.0\'\' E 46° 1\' N')).to.eql([567855.114, 96209.641]);
-      expect(getCoordinate(extent, '7° 1\' 25.0\'\' E 46° 1\' 25.0\'\' N')).to.eql([567859.21, 96981.625]);
-      expect(getCoordinate(extent, '46° 1\' 25.0\'\' N 7° 1\' 25.0\'\' E')).to.eql([567859.21, 96981.625]);
+    it('supports latitude and longitude as DMS', function(done) {
+      getCoordinate(extent, '7° E 46° N').then(function(position) {
+        expect(position).to.eql([566016.05, 94366.859]);
+        done();
+      });
+      $rootScope.$digest();
+      getCoordinate(extent, '7° 1\' E 46° N').then(function(position) {
+        expect(position).to.eql([567307.273, 94359.756]);
+        done();
+      });
+      $rootScope.$digest();
+      getCoordinate(extent, '7° 1\' 25.0\'\' E 46° N').then(function(position) {
+        expect(position).to.eql([567845.283, 94356.877]);
+        done();
+      });
+      $rootScope.$digest();
+      getCoordinate(extent, '7° 1\' 25.0\'\' E 46° 1\' N').then(function(position) {
+        expect(position).to.eql([567855.114, 96209.641]);
+        done();
+      });
+      $rootScope.$digest();
+      getCoordinate(extent, '7° 1\' 25.0\'\' E 46° 1\' 25.0\'\' N').then(function(position) {
+        expect(position).to.eql([567859.21, 96981.625]);
+        done();
+      });
+      $rootScope.$digest();
+      getCoordinate(extent, '46° 1\' 25.0\'\' N 7° 1\' 25.0\'\' E').then(function(position) {
+        expect(position).to.eql([567859.21, 96981.625]);
+        done();
```

Comment:
You can't use multiple time done functions in a `it()`  . Because  it consider the test good only if one `done()` is executed not all of them.

You need to use : 
```
$q.all([promise1, promise2, promise3]).then(function() {
  done()
}); 
```